### ISSUE TITLE
# character in osgeo password crashes the release

### ISSUE THREAD

**Issue by LKajan** [opengisch/qgis-plugin-ci#259]
When there is a '#' character in the password, the release process fails with the following error:
```
Traceback (most recent call last):
  File "/opt/hostedtoolcache/Python/3.[8](https://github.com/organization/repo/actions/runs/45645648/job/1345687894#step:5:9).18/x64/lib/python3.8/http/client.py", line 872, in _get_hostport
    port = int(host[i+1:])
ValueError: invalid literal for int() with base [10](https://github.com/organization/repo/actions/runs/45645648/job/1345687894#step:5:11): 'qwerty'
```
This occurs with a password 'qwerty#uiop'. It appears that parsing of the server address fails when there are special characters in password.

**Comment by LKajan** [comment]
Everything worked outside qgis-plugin-ci.
I found out that when the server address is parsed by the `ServerProxy` everything after the first '#' character was parsed as fragments and everything before came the actual netloc. Now when the netloc was tried to be splitted to host and port if ofcourse failed.
```python
>>> uri = 'https://username:qwe#rty@plugins.qgis.org:443/plugins/RPC2'
>>> urllib.parse.urlsplit(uri)     
SplitResult(scheme='https', netloc='username:qwe', path='', query='', fragment='rty@plugins.qgis.org:443/plugins/RPC2')
 ```
I provided a PR to fix this by forming the Basic authorization header manually.
